### PR TITLE
Drcandacemakedamoore/save as np

### PR DESCRIPTION
**Pull request by drcandacemakedamoore**
This adds a command-line option to save off csvs which were vertically oriented with respect to time as correctly oriented numpy files which can then be fed to the GUI. Still waiting to hear confirmation from user on this, but built as presumably many users may have this problem

### CODE REVIEW

File: resurfemg/converter_functions.py [hunk 1]
```diff
@@ -76,6 +78,71 @@ def csv_from_jkmn_to_array(file_name):
...
+    """
+    file = pd.read_csv(file_name)
+    new_df = (
+        file.T.reset_index().T.reset_index(drop=True)
+        .set_axis([f'lead.{i+1}' for i in range(file.shape[1])], axis=1)
+    )
+    arrayed = np.rot90(new_df)
+    arrayed = np.flipud(arrayed)
+    np.save(file_name, arrayed)
+    return arrayed
```
Comment (wbaccinelli) [inline resurfemg/converter_functions.py:143]: Do we want to have the function also returning the numpy array? If the user wants to have a converted file, probably it's not convenient to force him/her to also save it.

File: resurfemg/converter_functions.py [hunk 2]
```diff
@@ -76,6 +78,71 @@ def csv_from_jkmn_to_array(file_name):
...
+        arrayed = np.flipud(arrayed)
+
+        rel_fname = os.path.relpath(file_name, file_directory)
+        out_fname = os.path.join(made, rel_fname)
+        # check the directory does not exist
+        if not (os.path.exists(made)):
+            # create the directory you want to save to
+            os.mkdir(made)
+
+        np.save(out_fname, arrayed)
```
Comment (wbaccinelli) [inline resurfemg/converter_functions.py:118]: We can maybe reuse the code written in the `save_j_as_np_single`, like this:

```suggestion
        rel_fname = os.path.relpath(file_name, file_directory)
        out_fname = os.path.join(made, rel_fname)
        # check the directory does not exist
        if not (os.path.exists(made)):
            # create the directory you want to save to
            os.mkdir(made)
        save_j_as_np_single(file_name)
```

**Review by wbaccinelli — APPROVED** [review]
Looks good!

File: resurfemg/converter_functions.py (hunk 1 above)
Comment (drcandacemakedamoore) [reply resurfemg/converter_functions.py:143]: yes, but np.save() just has a side effect of a file, it doesn't return anything, and I'm wierded out by functions that don't return...although technically it should be fine...I'll think on it.

Comment (wbaccinelli) [reply resurfemg/converter_functions.py:143]: A possible approach could be to return something like the file path, which might be more relevant when using a saving function.

**Comment by drcandacemakedamoore** [comment]
This needs a rewrite now, because if we use Walter's suggestions it doesn't save to the right place

File: resurfemg/converter_functions.py (hunk 1 above)
Comment (drcandacemakedamoore) [reply resurfemg/converter_functions.py:143]: I guess we can wait to see how the user reacts, and then put in an issue to modify by what they want.

**Pull request merged**